src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -157,6 +157,15 @@ def check_non_empty_list_of_ints_in_range(min_value: int, max_value: int, max_si
     return check_list_of_ints_in_range(min_value, max_value, min_size=1, max_size=max_size, allow_null=allow_null)
 
 
+def check_no_duplicates() -> Callable:
+    def check_no_duplicates_checker(obj: Any):
```

Comment:
Is there a reason to return a new function object each time instead of having a single `check_no_duplicates` function and just using that in the validators list as `check_no_duplicates` (without parens)?

Reply:
it's just matching the style of the verifyers supplied earlier as a callable. if I do it this way, i can just iterate though all the verifiers and the previous ones can still have the parameters supplied.

Reply:
The iteration would work fine either way, unless I am missing something...  As in, the existing verifiers would not change.  But for this one, the actual Callable would be the function (which you would just list as the function name), not another function returned by calling the function.....

It probably does not matter much either way, but the unnecessary (?) indirection is a bit harder to read.

Reply:
nope, you're right - for this specific instance it's fine because it doesn't have parameters anymore. will change.